ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/VectorDBTool.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.tools;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.client.Client;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.ml.common.spi.tools.ToolAnnotation;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * This tool supports neural search with embedding models and knn index.
+ */
+@Log4j2
+@Getter
+@Setter
+@ToolAnnotation(VectorDBTool.TYPE)
+public class VectorDBTool extends AbstractRetrieverTool {
```

Comment:
This tool moved to skills https://github.com/opensearch-project/skills/blob/main/src/main/java/org/opensearch/agent/tools/VectorDBTool.java

Reply:
Updated. Is the plan to move all tools to skills repo?

Reply:
no, we plan to move tools which depend on other plugins to skills repo